webcomponents/tool/src/main/frontend/packages/sakai-picture-changer/src/SakaiPictureChanger.js

```diff
@@ -58,9 +59,16 @@ export class SakaiPictureChanger extends SakaiElement {
   _filePicked(e) {
 
     if (e.target.files[0]) {
-      this.cropper.clear();
-      this.cropper.replace(URL.createObjectURL(e.target.files[0]));
-      this._needsSave = true;
+      this._imageUrl = URL.createObjectURL(e.target.files[0]);
+      this.updateComplete.then(() => {
+        if (this.cropper) {
+          this.cropper.clear();
+          this.cropper.replace(this._imageUrl);
+        } else {
+          this._attachCropper();
+        }
+        this._needsSave = true;
```

Comment:
do we need to move `this._needsSave = true;` into the callback, I think it could stay where it was?

Reply:
I agree. I don't we need to move it. Thanks